uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -235,6 +235,10 @@ impl Enum {
         Ok(Literal::new_uint(this))
     }
 
+    pub fn variant_discr_type(&self) -> Option<Type> {
+        self.discr_type.clone()
```

Comment:
I think would work if you make this return `&Option<Type>` and avoid the clone, because the lifetime of the result should be the same as the `e` you are calling it on.